### PR TITLE
Add missing queries CLI

### LINKED ISSUE #1145: Add missing queries to CLI
## Summary

A number of queries were added in v1.4.5, specifically around unbonding and claims, for which cli commands were never added.

We should add the cli commands for all grpc queries.

### PR DESCRIPTION
**Pull request by tropicaldog**
## 1. Summary
Fixes #1145 

## 2.Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## 5. Checklist

- [x] Update tests

## Summary by CodeRabbit

- **New Features**
	- Introduced new query commands for enhanced user interaction with withdrawal records, redelegation records, validators, and transaction statuses.
	- Added functionality for users to retrieve their specific withdrawal records within a zone.

- **Improvements**
	- Improved the retrieval process of withdrawal and redelegation records to enhance performance and user experience.

- **Tests**
	- Implemented comprehensive tests for new functionalities, ensuring reliability and robustness of the system.

### CODE REVIEW

**Comment by tropicaldog** [comment]
Hi @joe-bowman,

I noticed that [`ZoneWithdrawalRecord`](https://github.com/quicksilver-zone/quicksilver/blob/main/x/interchainstaking/keeper/grpc_query.go#L240-L261) and [`UserWithdrawalRecord`](https://github.com/quicksilver-zone/quicksilver/blob/main/x/interchainstaking/keeper/grpc_query.go#L277-L292) seem to have identical functionality.

Given this, I was wondering if you think it might be worth refactoring these functions before I start working on issue #1233.
What are your thoughts on this?

**Comment by faddat** [comment]
I like the idea of combining the 2 lookalike functions, just please make sure they are exactly the same

File: x/interchainstaking/client/cli/cli_test.go [hunk 1]
```diff
@@ -66,6 +67,8 @@ func (s *IntegrationTestSuite) SetupSuite() {
 		MessagesPerTx:                0,
 		Is_118:                       true,
 	}
+
+	// TODO: I think setting validators here isn't enough, we need to set them in the store by using the keeper
```
Comment (joe-bowman) [inline x/interchainstaking/client/cli/cli_test.go:71]: Correct; we used to keep them in the zone struct but it was horribly inefficient loading 500 vals every time we wanted to fetch the denom from a zone for example.

**Comment by joe-bowman** [comment]
> Hi @joe-bowman,
> 
> I noticed that [`ZoneWithdrawalRecord`](https://github.com/quicksilver-zone/quicksilver/blob/main/x/interchainstaking/keeper/grpc_query.go#L240-L261) and [`UserWithdrawalRecord`](https://github.com/quicksilver-zone/quicksilver/blob/main/x/interchainstaking/keeper/grpc_query.go#L277-L292) seem to have identical functionality.
> 
> Given this, I was wondering if you think it might be worth refactoring these functions before I start working on issue #1233. What are your thoughts on this?

I don't think this is correct. ZoneWithdrawalRecords also scopes by chainId (perhaps ought to be called UserZoneWithdrawlRecords) whereas UserWithdrawalRecords doesn't. Would be good to see where these are used though?

**Comment by tropicaldog** [comment]
> I don't think this is correct. ZoneWithdrawalRecords also scopes by chainId (perhaps ought to be called UserZoneWithdrawlRecords) whereas UserWithdrawalRecords doesn't. Would be good to see where these are used though?

Yeah but still the name is kind of misleading. These aren't used any where beside grpc tests, so I think it's safe to refactor.

**Review by joe-bowman — APPROVED** [review]
lgtm

File: x/interchainstaking/keeper/grpc_query_test.go [hunk 2]
```diff
@@ -1411,3 +1411,130 @@
...
+			"UserZoneWithdrawalRecords_No_Withdrawal_Records",
+			func() {
+				zone, found := icsKeeper.GetZone(ctx, suite.chainB.ChainID)
+				tempAddr := addressutils.GenerateAccAddressForTest().String()
+				suite.True(found)
+
+				distributions := []*types.Distribution{
+					{
+						Valoper: icsKeeper.GetValidators(ctx, suite.chainB.ChainID)[0].ValoperAddress,
+						Amount:  10000000,
```
[the remark this answers is not in the document]
Comment (joe-bowman) [reply x/interchainstaking/keeper/grpc_query_test.go]: Ignore this; it's outdated. Appears fixed in the latest commit

**Pull request merged**